cmd/lnmuxd/log.go

```diff
@@ -1,16 +1,74 @@
 package main
 
 import (
+	"fmt"
+	"strings"
+
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
 var log *zap.SugaredLogger
 
-func init() {
-	config := zap.NewDevelopmentConfig()
-	config.EncoderConfig.EncodeLevel = zapcore.CapitalColorLevelEncoder
-	config.EncoderConfig.EncodeCaller = nil
-	rawLog, _ := config.Build()
-	log = rawLog.Sugar()
+// initLogger constructs a new SugaredLogger.
+// By default, the level is 'debug', the format is 'console' and it doesn't display the caller.
+func initLogger(level string, format string, withCaller bool) error {
+	var l zapcore.Level
+	switch strings.TrimSpace(strings.ToLower(level)) {
+	case "", "d", "debug":
+		l = zap.DebugLevel
+	case "info", "i":
+		l = zap.InfoLevel
+	case "w", "warn", "warning":
+		l = zap.WarnLevel
+	case "e", "error", "errors":
+		l = zap.ErrorLevel
+	case "dp", "dpanic":
+		l = zap.DPanicLevel
+	case "p", "panic", "panics":
+		l = zap.PanicLevel
+	case "f", "fatal", "fatals":
+		l = zap.FatalLevel
+	default:
+		return fmt.Errorf("unknown level %s", level)
+	}
+
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	encoderConfig.TimeKey = "time"
+
+	if !withCaller {
+		encoderConfig.EncodeCaller = nil
+	}
+
+	// format is 'console' by default
+	if format == "" {
+		// Add color to the level
+		encoderConfig.EncodeLevel = zapcore.CapitalColorLevelEncoder
+		format = "console"
+	}
+
+	config := zap.Config{
+		Level:       zap.NewAtomicLevelAt(l),
+		Development: false,
+		Sampling: &zap.SamplingConfig{
+			Initial:    100,
+			Thereafter: 100,
+		},
+		Encoding:         format,
+		EncoderConfig:    encoderConfig,
+		OutputPaths:      []string{"stderr"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+
+	lo, err := config.Build()
+	if err != nil {
+		return err
+	}
+
+	lo = lo.With(zap.String("svc", "lnmux"))
```

Comment:
Isn't this unnecessary as there are no other services in this process?

Reply:
It is interesting to have this, so we can filter logs given the service name (in an environment where we have various services).

Reply:
Isn't a field for the service name automatically added in such environments by the environment itself? I mean, in k8s for example, not all logs line of all processes are piled on a single heap with any additional identification of where it originates from?

Reply:
Yeah, I guess we can filter per container name in k8s. But maybe we will have some other environment where the service name will be missing? 
I don't really know about other env.
 K8s remains the norm so I can remove it I guess!